src/World.cpp

```diff
@@ -1039,14 +1068,123 @@ void cWorld::Tick(std::chrono::milliseconds a_Dt, std::chrono::milliseconds a_La
 
 	if (m_WorldAge - m_LastSave > std::chrono::minutes(5))  // Save each 5 minutes
 	{
-		SaveAllChunks();
+		// LOGD("%s: ********* Saving ALL dirty chunks (periodic save)", GetName().c_str());
+		SaveChunks();
+	}
+}
+
+
+
+
+
+int cWorld::ManageChunkUnload()
+{
+	/* #ifdef _DEBUG
+	size_t MyRam = GetMemoryCounter().GetApproximateChunkRAM();  // RAM used by our loaded chunks (approximated)
+	#endif */
+
+	cRoot * Root = cRoot::Get();
+	size_t Ram = Root->GetApproximateChunkRAM();  // RAM used by all world's loaded chunks combined (approximated) in MiB
+	size_t MaxRam = Root->GetMaxRAMChunks();      // Maximum allowed RAM use of all worlds in MiB
+
+	LOGD("%s: GlobalRam: %zu, myChunks: %zu, Unloadable: %zu", GetName().c_str(), Ram, GetNumChunks(), GetMemoryCounter().GetCanUnloadCount());
+
+	// If the setting is set to minimize ram
+	if (Root->GetMinimizeRam())
+	{
+		// Save all chunks if RAM is too much
+		if (Ram > MaxRam)
+		{
+			if (m_WorldAge - m_LastSave > std::chrono::seconds(20))
+			{
+				LOGD("###### %s: Saving all unused dirty chunks", GetName().c_str());
+				SaveChunks(true);
```

Comment:
This is more efficient than saving all chunks, because it only saves chunks which can be unloaded. But if the server crashes, then only some chunks would have been saved. Should I go for `SaveChunks();` (Save them all) to avoid this until Cuberite becomes more stable?